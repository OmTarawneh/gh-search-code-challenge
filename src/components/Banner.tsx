"use client";

import type { FC } from "react";
import Typography from "@mui/material/Typography";

const Banner: FC = () => {
	return <Typography variant="h1">GitHub Search</Typography>;
};

export default Banner;
