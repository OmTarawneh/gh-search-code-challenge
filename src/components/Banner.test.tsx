// Banner.test.js

import React from "react";
import { render } from "@testing-library/react";
import Banner from "./Banner";

describe("Banner", () => {
	it("renders the component with the correct text", () => {
		const { getByText } = render(<Banner />);
		const bannerText = getByText("GitHub Search");

		// Ensure that the component is rendered with the correct text
		expect(bannerText).toBeTruthy();
	});

	it("matches the snapshot", () => {
		const { container } = render(<Banner />);

		// Check if the rendered component matches the snapshot
		expect(container).toMatchSnapshot();
	});
});
