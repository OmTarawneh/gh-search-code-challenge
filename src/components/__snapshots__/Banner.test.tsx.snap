// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Banner matches the snapshot 1`] = `
<div>
  <h1
    class="MuiTypography-root MuiTypography-h1 css-o2w69a-MuiTypography-root"
  >
    GitHub Search
  </h1>
</div>
`;
